flink-ml-lib/src/test/java/org/apache/flink/ml/classification/knn/KnnTest.java

```diff
@@ -0,0 +1,251 @@
+            Table output, String labelCol, String predictionCol) throws Exception {
+        StreamTableEnvironment tEnv =
+                (StreamTableEnvironment) ((TableImpl) output).getTableEnvironment();
+
+        DataStream<Tuple2<Double, Double>> stream =
+                tEnv.toDataStream(output)
+                        .map(
+                                new MapFunction<Row, Tuple2<Double, Double>>() {
+                                    @Override
+                                    public Tuple2<Double, Double> map(Row row) {
+                                        return Tuple2.of(
+                                                (Double) row.getField(labelCol),
+                                                (Double) row.getField(predictionCol));
+                                    }
+                                });
+        return IteratorUtils.toList(stream.executeAndCollect());
+    }
+
+    private static void verifyClusteringResult(List<Tuple2<Double, Double>> result) {
+        for (Tuple2<Double, Double> t2 : result) {
+            Assert.assertEquals(t2.f0, t2.f1);
+        }
+    }
+
+    @Test
+    public void testParam() {
+        Knn knn = new Knn();
+        assertEquals("features", knn.getFeaturesCol());
+        assertEquals("label", knn.getLabelCol());
+        assertEquals(10L, knn.getK().longValue());
+        assertEquals("prediction", knn.getPredictionCol());
+
+        knn.setLabelCol("test_label")
+                .setFeaturesCol("test_features")
+                .setK(4)
+                .setPredictionCol("test_prediction");
+
+        assertEquals("test_features", knn.getFeaturesCol());
+        assertEquals("test_label", knn.getLabelCol());
+        assertEquals(4L, knn.getK().longValue());
+        assertEquals("test_prediction", knn.getPredictionCol());
+    }
+
+    @Test
+    public void testFeaturePredictionParam() throws Exception {
+        Knn knn =
+                new Knn()
+                        .setLabelCol("test_label")
+                        .setFeaturesCol("test_features")
+                        .setK(4)
+                        .setPredictionCol("test_prediction");
+        KnnModel model = knn.fit(trainData.as("test_features, test_label"));
+        Table output = model.transform(testData.as("test_features, test_label"))[0];
+
+        assertEquals(
+                Arrays.asList("test_features", "test_label", "test_prediction"),
+                output.getResolvedSchema().getColumnNames());
+
+        List<Tuple2<Double, Double>> result =
+                executeAndCollect(output, "test_label", "test_prediction");
+        verifyClusteringResult(result);
+    }
+
+    @Test
+    public void testFewerDistinctPointsThanCluster() throws Exception {
+        Knn knn = new Knn().setK(4);
+        KnnModel model = knn.fit(testData);
+        Table output = model.transform(testData)[0];
+        executeAndCollect(output, "label", "prediction");
```

Comment:
can we replace `executeAndCollect(output, "label", "prediction");` with `executeAndCollect(output, knn.getLabelCol(), knn.getPredictionCol())` here?

Reply:
OK